packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1748,15 +1777,16 @@ class LeaderLayer extends ContainerLayer {
   Iterable<S> findAll<S>(Offset regionOffset) => super.findAll<S>(regionOffset - offset);
 
   @override
-  ui.EngineLayer addToScene(ui.SceneBuilder builder, [ Offset layerOffset = Offset.zero ]) {
+  void addToScene(ui.SceneBuilder builder, [ Offset layerOffset = Offset.zero ]) {
     assert(offset != null);
     _lastOffset = offset + layerOffset;
     if (_lastOffset != Offset.zero)
-      builder.pushTransform(Matrix4.translationValues(_lastOffset.dx, _lastOffset.dy, 0.0).storage);
+      engineLayer = builder.pushTransform(Matrix4.translationValues(_lastOffset.dx, _lastOffset.dy, 0.0).storage, oldLayer: engineLayer);
```

Comment:
Yes, when a leader/follower layer is moving without changing its children, we want to reuse the engine layer DOM nodes associated with `pushTransform`.

Reply:
I see. Maybe add a comment somewhere like "The engine layer's resources may be reused through oldLayer, but do not reuse the engine layer's rasterized pixels through addRetained directly because there's no guarantee that the pixels would stay the same even if the whole subtree is unchanged."

Reply:
This is specific to the Leader/FolllowerLayer. Previously, it's impossible to re-use the engineLayer of a Leader/FollowerLayer for `addRetained` because we return null.

Now we return the engineLayer for oldLayer, but we still don't want developers to use it for `addRetained`. I don't think that we need to document every `push*` method. Just documenting `Leader/FollowerLayer.addToScene` would be enough. Maybe something like "Only use the returned engineLayer for oldLayer in .... Do not use it for addRetained".